drivers/ethernet/eth_nxp_enet.c

```diff
@@ -744,57 +746,6 @@ static const struct ethernet_api api_funcs = {
 #error "Unsupported SOC"
 #endif
 
-#define NXP_ENET_GENERATE_MAC_RANDOM(n)						\
-	static void generate_eth_##n##_mac(uint8_t *mac_addr)			\
-	{									\
-		gen_random_mac(mac_addr,					\
-			       FREESCALE_OUI_B0,				\
-			       FREESCALE_OUI_B1,				\
-			       FREESCALE_OUI_B2);				\
-	}
-
-#define NXP_ENET_GENERATE_MAC_UNIQUE(n)						\
-	static void generate_eth_##n##_mac(uint8_t *mac_addr)			\
-	{									\
-		uint32_t id = ETH_NXP_ENET_UNIQUE_ID;				\
```

Comment:
I agree, I deleted it on accident, but, it needs to have a new DT property introduced then because, as discussed on some other open tickets to ethernet drivers, this behavior is not part of the common zephyr/DT API, it either needs to be documented that the default behavior if a node has no Mac address is to use a pretermined one, or there needs to be another explicit DT property to clarify it. Right now a lot of drivers leave this to completely undefined behavior or even explicitly statically assert this case and cause a build error if the mac address is not specified in DT.

So the previous behavior is not standard and cannot be kept as it is... but I will definitely address adding the feature somewhat properly in this PR. Although I will not take any approach that changes the common ethernet interface in DT or documentation regarding this in this PR as that would require ensuring it to be true for all other ethernet drivers which is a scope creep.

Reply:
To clarify, what I am going to just do for now in this PR is introduce a boolean option in the nxp,enet binding to accomplish this, in a followup PR I will investigate a standard way to do it since it is a common feature and this keeps repearedly coming up as a point of debate on other ethernet driver PRs as well

Reply:
Also, the fact that the old code is setting the LAA bit seems incorrect to me, since what we are literally doing is the description of a UAA address, ie, using the vendor OUI and then setting the rest based on some unique silicon id. The only controversy about is we are deciding this in OS software rather than simply reading the entire address from some ROM. I still think it's not right to set the LAA bit here, but I'm not sure.